presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -420,9 +421,15 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
 
                 leftSource = new ProjectNode(idAllocator.getNextId(), leftSource, leftProjections.build());
                 rightSource = new ProjectNode(idAllocator.getNextId(), rightSource, rightProjections.build());
-
+                PlanNodeId id;
+                if (node.getDynamicFilterAssignments().size() > 0) {
```

Comment:
don't rely on `nodeId`. It doesn't have much semantics. Instead `dynamicFilterSourceId` is needed in `JoinNode` as i've suggested in other comment.

Reply:
Removed.